cuking.cu

```diff
@@ -230,7 +231,10 @@ __global__ void ComputeKingKernel(
     num_het_j += __popcll(het_j & missing_mask);
     num_both_het += __popcll(het_i & het_j & missing_mask);
     num_opposing_hom += __popcll(
-        ((hom_ref_i & hom_alt_j) | (hom_ref_j & hom_alt_i)) & missing_mask);
+        ((hom_ref_i & hom_alt_j) | (hom_alt_i & hom_ref_j)) & missing_mask);
+    num_concordant_hom += __popcll(
+        ((hom_ref_i & hom_ref_j) | (hom_alt_i & hom_alt_j)) & missing_mask);
+    num_shared_sites += __popcll(missing_mask);
```

Comment:
(can't comment on the lines that are not changed in the PR, but wanted to comment on `missing_mask`)

```
const uint64_t missing_mask = ~(het_i & hom_alt_i) & ~(het_j & hom_alt_j);
```

It's a bit hard to understand. Maybe remind here again in a comment that `het_i` and `hom_alt_i` being both set to `1` is a code for a missing genotype?

And maybe rename `missing_mask` to `defined` as it seems to be `1` when a genotype is _not_ missing?

Feel free to ignore, just a minor nit picking.

Reply:
Great points, thanks! The comment down at line 517 isn't really helpful when trying to understand this logic :). I've also renamed the variable as you suggested, to make this less confusing.